tiledb/sm/array_schema/array_schema.cc

```diff
@@ -1811,3 +1761,52 @@ void ArraySchema::generate_uri(
 }
 
 }  // namespace tiledb::sm
+
+std::ostream& operator<<(
+    std::ostream& os, const tiledb::sm::ArraySchema& schema) {
+  os << "- Array type: " << array_type_str(schema.array_type()) << "\n";
+  os << "- Cell order: " << layout_str(schema.cell_order()) << "\n";
+  os << "- Tile order: " << layout_str(schema.tile_order()) << "\n";
+  os << "- Capacity: " << schema.capacity() << "\n";
+  os << "- Allows duplicates: " << (schema.allows_dups() ? "true" : "false")
+     << "\n";
+  os << "- Coordinates filters: " << schema.coords_filters().size();
```

Comment:
It would be better for the output to always end in a newline.

Reply:
This is done in `os << schema.coords_filters();` which follows.